packages/flutter/lib/src/services/keyboard_key.g.dart

```diff
@@ -225,35 +225,40 @@ class LogicalKeyboardKey extends KeyboardKey {
 
   /// Returns a set of pseudo-key synonyms for the given `key`.
   ///
-  /// This allows finding the pseudo-keys that also represents a concrete
-  /// `key` so that a class with a key map can match pseudo-keys as well as the
-  /// actual generated keys.
-  ///
-  /// The pseudo-keys returned in the set are typically used to represent keys
-  /// which appear in multiple places on the keyboard, such as the [shift],
-  /// [alt], [control], and [meta] keys. The keys in the returned set won't ever
-  /// be generated directly, but if a more specific key event is received, then
+  /// This allows finding the pseudo-keys that also represent a concrete `key`
+  /// so that a class with a key map can match pseudo-keys as well as the actual
+  /// generated keys.
+  ///
+  /// Pseudo-keys returned in the set are typically used to represent keys which
+  /// appear in multiple places on the keyboard, such as the [shift], [alt],
+  /// [control], and [meta] keys. Pseudo-keys in the returned set won't ever be
+  /// generated directly, but if a more specific key event is received, then
   /// this set can be used to find the more general pseudo-key. For example, if
   /// this is a [shiftLeft] key, this accessor will return the set
   /// `<LogicalKeyboardKey>{ shift }`.
-  Set<LogicalKeyboardKey> get synonyms {
-    final LogicalKeyboardKey? result = _synonyms[this];
-    return result == null ? <LogicalKeyboardKey>{} : <LogicalKeyboardKey>{result};
-  }
+  Set<LogicalKeyboardKey> get synonyms => _synonyms[this] ?? <LogicalKeyboardKey>{};
 
   /// Takes a set of keys, and returns the same set, but with any keys that have
   /// synonyms replaced.
   ///
-  /// It is used, for example, to make sets of keys with members like
+  /// It is used, for example, to take sets of keys with members like
   /// [controlRight] and [controlLeft] and convert that set to contain just
   /// [control], so that the question "is any control key down?" can be asked.
   static Set<LogicalKeyboardKey> collapseSynonyms(Set<LogicalKeyboardKey> input) {
-    final Set<LogicalKeyboardKey> result = <LogicalKeyboardKey>{};
-    for (final LogicalKeyboardKey key in input) {
-      final LogicalKeyboardKey? synonym = _synonyms[key];
-      result.add(synonym ?? key);
-    }
-    return result;
+    return input.expand((LogicalKeyboardKey element) {
+      return _synonyms[element] ?? <LogicalKeyboardKey>{element};
+    }).toSet();
+  }
+
+  /// Returns the given set with any pseudo-keys expanded into their synonyms.
+  ///
+  /// It is used, for example, to take sets of keys with members like [control]
+  /// and [shift] and convert that set to contain [controlLeft], [controlRight],
+  /// [shiftLeft], and [shiftRight].
+  static Set<LogicalKeyboardKey> expandSynonyms(Set<LogicalKeyboardKey> input) {
+    return input.expand((LogicalKeyboardKey element) {
+      return _reverseSynonyms[element] ?? <LogicalKeyboardKey>{element};
```

Comment:
I think this logic is so simple that it should probably be written as a `switch` clause whose `default` is `assert(false)`.

Reply:
I'm not quite sure what you mean: do you mean removing the `_reverseSynonyms` map and just hard coding it here? Or do you mean something else?

For example:
```dart
  return input.expand((LogicalKeyboardKey key) {
    return switch(key) {
      LogicalKeyboardKey.shift => <LogicalKeyboardKey>[LogicalKeyboardKey.shiftLeft, LogicalKeyboardKey.shiftRight],
      // ...
      _ => assert(false),
    } as Iterable<LogicalKeyboardKey>;
  });
```

Reply:
Yeah. I think saving a map is clearer. But it's up to you.

Also, do we expect this function to be used by others, or is it just for internal variables like `_controlSynonyms`? If the latter, then we might as well just use `_ => assert(false)` instead of `_ => <LogicalKeyboardKey>[key]`.

Reply:
I was writing it expecting it to be generally useful. If it was just for internal variables, I'd just hard code those as private lists. I wanted to have the internal variables depend on the generated file, though, so that it wouldn't need to be updated manually if the list of synonyms gets added to. And if I wanted that, then anyone else needing this functionality might also want to do that.

Reply:
Got it, ok, let's keep the current approach.